apps/gdalalg_raster_color_merge.h

```diff
@@ -0,0 +1,59 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  "color-merge" step of "raster pipeline"
+ * Author:   Even Rouault <even dot rouault at spatialys.com>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Even Rouault <even dot rouault at spatialys.com>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#ifndef GDALALG_RASTER_COLOR_MERGE_INCLUDED
+#define GDALALG_RASTER_COLOR_MERGE_INCLUDED
+
+#include "gdalalg_raster_pipeline.h"
+
+//! @cond Doxygen_Suppress
+
+/************************************************************************/
+/*                  GDALRasterColorMergeAlgorithm                       */
+/************************************************************************/
+
+class GDALRasterColorMergeAlgorithm /* non final*/
+    : public GDALRasterPipelineStepAlgorithm
+{
+  public:
+    explicit GDALRasterColorMergeAlgorithm(bool standaloneStep = false);
+
+    static constexpr const char *NAME = "color-merge";
+    static constexpr const char *DESCRIPTION =
+        "Replace the color value of a RGB/RGBA dataset by the one of a "
+        "grayscale raster";
```

Comment:
This description does not make much sense to me Isn't the direction rather the opposite, from RGB to what-used-to-be grayscale?

Reply:
would "Use a grayscale raster to replace the intensity of a RGB/RGBA dataset" make more sense ?  (knowing that "intensity" is a bit improper here as it is actually the V(alue) component in HSV color space, but intensity should be a reasonable quick summary)

Reply:
For me, yes. But some regular GIS user may think that it is the hillshade image that gets colored, not the RGB image that gets intensified.

Reply:
> But some regular GIS user may think that it is the hillshade image that gets colored, not the RGB image that gets intensified.

isn't that the same thing: replace (hue, saturation) of the hillshade with (hue, saturation) of the RGB   vs replace "value" (intensity) of the RGB with value of the hillshade ?

I'm open to better wording if you can come with something :-)

Reply:
I think it is good enough. And you described one use case already as "This can be for example to colorize a hillshade raster generated by gdal_raster_hillshade with an hypsometric rendering of a DEM".